src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -82,6 +83,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 	String replacement = "<img src=\"$1\" />";
 	String textWithImagesReplaced = userText.replaceAll(regex, replacement);
 
+	int i = 0;
+	String[] schemes = {"http","https"};
+	UrlValidator urlValidator = new UrlValidator(schemes);
+	while (true) {
```

Comment:
This loop should work but it's a bit hard to follow at first glance, do you think we might be able to use regex matching to extract the URLs and make it a bit more easily readable?

Reply:
Were you thinking like creating a regex string to find URLs, adding them to a list and then iterating through the list verifying them?

Reply:
Yep that was exactly what I was thinking, using the Regex Matcher's "find" function would be a good way to structure the loop as "for each URL" which I think would be followed more easily.

Reply:
Updated to use regex! Let me know what you think.